connector-sdk/core/src/main/java/io/camunda/connector/api/error/ConnectorCorrelationException.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.connector.api.error;
+
+/** A retryable exception indicating failed inbound event correlation */
+public class ConnectorCorrelationException extends ConnectorException {
+
+  public enum CorrelationErrorReason {
+    FAULT_ZEEBE_CLIENT_STATUS,
```

Comment:
Would it work for you if we remove the error reason completely and just wrap the original error into plain `ConnectorCorrelationException` for easier catching on the connector side (let's say if it we want to catch it not immediately but rather in downstream calls where it may be confused with other exceptions)?

Reply:
I saw we pass the original exception as well. So we are good.

Reply:
@chillleader Yes, also reasonable.